packages/frontend/src/components/staking/components/Validator.js

```diff
@@ -98,25 +98,28 @@ export default function Validator({
         setShowClaimTokenFarmRewardsModal(true);
     };
 
+    const isFarmingValidator = validator?.version === FARMING_VALIDATOR_VERSION;
+
     const handleStakeAction = async () => {
         if (showConfirmModal && !loading) {
-            await StakingFarmContracts.getFarmListWithUnclaimedRewards({
-                contractName: validator.contract.contractId,
-                account_id: currentAccountId,
-                from_index: 0,
-                limit: 300,
-            }).then((res) => 
-                Promise.all([
-                    (res || [])
-                    .filter(({balance}) => !new BN(balance).isZero())
-                    .map(({token_id}) => dispatch(claimFarmRewards(validator.accountId, token_id)))
-                ])
-            );
+            if (isFarmingValidator) {
+                await StakingFarmContracts.getFarmListWithUnclaimedRewards({
```

Comment:
Not a blocker for this, but we should fix this up in a follow-up -- since `claim()` takes `token_id` as an argument, we only need to call this once per `token_id` that _any_ farm on the contract yields.

Reply:
Added to WAL-213